cmake/SetupCodeChecks.cmake

```diff
@@ -206,8 +217,121 @@ macro(blt_add_code_checks)
                                  SRC_FILES         ${_c_sources})
     endif()
 
+    if (CLANGQUERY_FOUND)
+        set(_clang_query_target_name ${arg_PREFIX}_clang_query_check)
+        blt_error_if_target_exists(${_clang_query_target_name} ${_error_msg})
+        blt_add_clang_query_target( NAME           ${_clang_query_target_name}
+                                 WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+                                 SRC_FILES         ${_c_sources})
+    endif()
+
 endmacro(blt_add_code_checks)
 
+##-----------------------------------------------------------------------------
+## blt_configure_clang_query( CLANG_QUERY_DIR
+##                          )
+##
+## Configures the clang query helper script and finds clang query
+##
+##-----------------------------------------------------------------------------
+macro(blt_configure_clang_query)
+    set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
```

Comment:
I removed this macro entirely, with the move to a non-configured script it became unnecessary. 

Setting CMAKE_EXPORT_COMPILE_COMMANDS to on still happens (now in blt_add_clang_query_target, which is questionable, as it doesn't seem to universally produce the file. May be one of those CMake flags that has to be set on the cmake line). We might want to just document "hey, you need to invoke CMake with this argument"

Reply:
I wonder if it is not working because cmake already adds it to the cache (somewhere) with a value of ``OFF``?  Or if it is being called too late...

Could we move it higher, e.g. around here: https://github.com/LLNL/blt/blob/master/SetupBLT.cmake#L104 ?

The note in the cmake docs claims it only works for make and ninja, and is harmless elsewhere -- https://cmake.org/cmake/help/latest/variable/CMAKE_EXPORT_COMPILE_COMMANDS.html